app/services/fact_cleaner.rb

```diff
@@ -1,80 +1,58 @@
 class FactCleaner
   delegate :logger, :to => :Rails
 
+  def initialize(batch_size: 500)
+    @batch_size = batch_size
+  end
+  
   def clean!
-    @deleted_count = delete_excluded_facts + delete_orphaned_facts
+    @deleted_count = 0
+    delete_excluded_facts(@batch_size)
```

Comment:
Do we need to pass `@batch_size` explicitly? We can just use it inside the inner methods.